lib/internal/Magento/Framework/Validator/EmailAddress.php

```diff
@@ -6,36 +6,9 @@
  * See COPYING.txt for license details.
  */
 namespace Magento\Framework\Validator;
+use Laminas\Validator\EmailAddress as LaminasEmailValidator;
 
-class EmailAddress extends \Zend_Validate_EmailAddress implements \Magento\Framework\Validator\ValidatorInterface
+class EmailAddress extends LaminasEmailValidator
```

Comment:
Hi @konarshankar07. Thank you for your work. Could you explain your fix, please? I see that you have removed the part that disables TLD validation. But validation of the name in the email (before @) still in place. 

Also, you've removed the interface mention from the validator. Could you shed some light on a reason, please? 

Thanks.

Reply:
Hello @rogyar 
I think we should remove the zendframework1 dependency from the magento and use the Laminas dependency to the magento core that's why I removed the interface and TLD validation and extended the laminas Email validator.Let me know your thoughts
Thanks

Reply:
Hi @konarshankar07. That makes sense, thank you for the explanation. 

In the current version, we have the TLD validation set to `false`. However, in Laminas hostname validator, the `useTldCheck` set to `true` by default. So it means that after your change the TLD will be validated (but should not). 

Also, still not obvious how this change will fix validation of the `THIS@mail.com` part of the email address that may contain diacritics. 

Could you clarify the mentioned points, please? 

Thanks.